.github/workflows/on-pull-request.yml

```diff
@@ -1,8 +1,6 @@
 name: Run checks on Pull Requests
 on:
   pull_request:
-    branches:
-      - main
```

Comment:
I think we should remove this because devs might be stacking PRs as I did, and while recursively merging will eventually catch any CI failures from upstack changes, my sense is they'd want to know before merging if their branch fails unit tests. 

Lmk if there's any $ costs to be mindful of, and if this makes sense

Reply:
oh this is a good catch!